query/src/common/expression_evaluator.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_datablocks::DataBlock;
+use common_datavalues::ColumnRef;
+use common_datavalues::ColumnWithField;
+use common_datavalues::DataField;
+use common_datavalues::DataType;
+use common_datavalues::DataTypeImpl;
+use common_exception::ErrorCode;
+use common_exception::Result;
+use common_functions::scalars::CastFunction;
+use common_functions::scalars::FunctionContext;
+use common_functions::scalars::FunctionFactory;
+use common_planners::Expression;
+
+use crate::sessions::QueryContext;
+
+pub struct ExpressionEvaluator;
+
+impl ExpressionEvaluator {
+    pub fn eval(
+        expression: &Expression,
+        block: &DataBlock,
+        ctx: &QueryContext,
```

Comment:
'ctx' is generally used as the first parameter of the function

Reply:
Fixed.